### ISSUE TITLE
TVTensor Mask and Image class names are incorrect

### ISSUE THREAD

**Issue by sebsken** [pytorch/vision#8547]
### 🐛 Describe the bug

The TV tensor class/type "Mask" is incorrect as in the case of image matting, the mask would be a gray level image.
There are also cases where non "mask" content should use this class such as for any class related content.

The name of this class could be "Indexed" instead as this is really what this kind of tensor is: class indexes, color indexes, or other. Basically, content with a range of valid values limited to the ones already present in it and whose meaning is unrelated to their numerical value.

Similarly, the "Image" TVTensor name could be improved as this kind of tensor is also valid for depth maps for instance. It could be "Raster" or "Raw" or "Continuous". Raster and Raw are heavily image connotated but more generic (and this is a torchvision library anyways). Continuous can be confusing especially in the case of integer data type although the meaning is correct within the appropriate data type.

### Versions

Collecting environment information...
PyTorch version: 2.2.0+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)

Python version: 3.12.1 (main, Dec 10 2023, 15:16:41) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
Is CUDA available: True

Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] torch==2.2.0
[pip3] torchvision==0.17.0
[pip3] triton==2.2.0
[conda] Could not collect

**Comment by NicolasHug** [comment]
Thanks for the suggestions @sebsken . `Mask` was named `Mask` because in the context of torchvision, those correspond to segmentation or detection masks.

They can of course be extended to support more than just these type of objects, and I'm glad to hear that you find them useful in other contexts.

It is of course impossible to find naming that would fit every single existing use-case. What we recommend is for users to simply inherit these base classes in their code-base to more descriptive names. This allows everyone to fine-tune their approach and allow us to keep the names in torchvision both generic *and* accurate.
